app/uk/gov/hmrc/customs/declaration/connectors/upscan/UpscanInitiateConnector.scala

```diff
@@ -54,7 +54,7 @@ class UpscanInitiateConnector @Inject()(http: HttpClient,
         res
       }
       .recoverWith {
-        case httpError: HttpException => Future.failed(new RuntimeException(httpError))
```

Comment:
Unnecessary wrapping that prevents destructuring in `recoverWith` case statement.